font/src/ft/mod.rs

```diff
@@ -503,6 +560,9 @@ impl FreeTypeRasterizer {
                         // load it again.
                         Some(&key) => {
                             debug!("Hit for font {:?}; no need to load", path);
+                            // Update fixup factor
+                            self.faces.get_mut(&key).unwrap().pixelsize_fixup_factor =
+                                pattern.pixelsizefixupfactor().next().unwrap_or(0.0);
```

Comment:
You can change font size and it'll break things.

Reply:
But how do we change the font size without reloading the font?

I don't think there's currently any way, unless I'm missing things.

Reply:
You're right about reloading, but I guess you're thinking only about alacritty.yml fonts, and this will be fine for them, but we also have a font fallbacks, so to handle them properly you either need to load an entire font during fallback and keep it around (I like this idea though) or do the same thing as I do here. The only way to get `fixup_factor`  is from `font_match` resulted pattern. So you have 3 places in `ft/mod.rs` 2 of them are about default config and 1 for fallback, so you should perform the update in a fallback part.

Reply:
Okay, here's my understanding of the problem, let me know if that is correct:

 - We load our default font into `self.faces`
 - Due to a missing emoji glyph, we load a fallback font into `self.faces`
 - We resize our default font, effectively discarding it from `self.faces`
    -> To be completely correct, I don't think we ever actually discard the old face? I think we just insert the new one and the old one is just never accessed anymore
 - We fallback to the same emoji font again, but it has not been 'overwritten' and the same font matches both our old size pattern as the new one, thus getting loaded from `self.faces` with incorrect fixup factor

So my initial thought would almost be that we might want to clear `self.faces` whenever a new default font is loaded? However that does not seem ideal for performance, but it will happen much less than font fallback in general.

I'd be curious what you think, though generally I feel like this should be our safest choice from a performance standpoint, since the operation is basically a noop with how cold it is.

Reply:
You're correct here. However I don't like the idea of dropping things and reloading them once again in the current state.

The main problem with this approach is that we should provide a new API to clear Rasterizer "cache" and it doesn't make much sense with the current font crate, since you'll "pseudo fallback" anyway most of the time. However if we do our fallbacks without using fontconfig api all the time (Like load fonts once and don't call it unless we can't find anything in already loaded fonts) that'll make much more sense and improve overall performance, since now e.g. unicode-random-write is a fontcofnig (`fc-match`) benchmark, which is in the same time funny and not.

Anyway this rework is out of scope of "Add colored emoji PR" in my opinion.